Unstructured-IO/unstructured-api
# chore: Add more env variables to tune parallel mode

Set the number of fanout threads and the pdf split size via env variable.

Test instructions:
* Enable parallel mode and run locally:
```
export UNSTRUCTURED_PARALLEL_MODE_URL=http://localhost:8000/general/v0/general
export UNSTRUCTURED_PARALLEL_MODE_ENABLED=true
export UNSTRUCTURED_PARALLEL_MODE_SPLIT_SIZE=1
export UNSTRUCTURED_PARALLEL_MODE_THREADS=3
make run-web-app
```

I'd suggest using hi_res mode and watch the logs for the expected behavior. 

```
curl -X 'POST' 'http://localhost:8000/general/v0/general' -F 'files=@sample-docs/layout-parser-paper.pdf' -F 'strategy=hi_res'
```

When we process a page you'll see
```
2023-06-29 16:56:47,889 unstructured_inference INFO Detecting page elements ...
```

So, to verify:
* Change the thread count to 5 and verify that you see 5 concurrent inference messages
* Change thread count to 2 and split size to 5, verify that you see multiple inferences as pages are processed, followed by two fastapi logs at a time as responses are returned.
